From 93fc5644f60bf2c5d7bfa4aa5525e6888f28bfff Mon Sep 17 00:00:00 2001
From: Gary Zhang <b13634@freescale.com>
Date: Wed, 18 Apr 2012 12:38:37 +0800
Subject: ENGR00180075 MX6: change CLKO source to pll4_audio_main_clk

change CLKO source to pll4_audio_main_clk for low power
mode consideration

Signed-off-by: Gary Zhang <b13634@freescale.com>
---
 arch/arm/mach-mx6/clock.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-mx6/clock.c b/arch/arm/mach-mx6/clock.c
index 3b815fe..20cfc55 100644
--- a/arch/arm/mach-mx6/clock.c
+++ b/arch/arm/mach-mx6/clock.c
@@ -5180,8 +5180,8 @@ int __init mx6_clocks_init(unsigned long ckil, unsigned long osc,
 	sata_clk[0].disable(&sata_clk[0]);
 	pcie_clk[0].disable(&pcie_clk[0]);
 
-	/* Initialize Audio and Video PLLs to valid frequency (650MHz). */
-	clk_set_rate(&pll4_audio_main_clk, 650000000);
+	/* Initialize Audio and Video PLLs to valid frequency. */
+	clk_set_rate(&pll4_audio_main_clk, 176000000);
 	clk_set_rate(&pll5_video_main_clk, 650000000);
 
 	clk_set_parent(&ipu1_di_clk[0], &pll5_video_main_clk);
@@ -5226,7 +5226,7 @@ int __init mx6_clocks_init(unsigned long ckil, unsigned long osc,
 	clk_set_parent(&clko2_clk, &osc_clk);
 	clk_set_rate(&clko2_clk, 2400000);
 
-	clk_set_parent(&clko_clk, &ipg_clk);
+	clk_set_parent(&clko_clk, &pll4_audio_main_clk);
 	/*
 	 * FIXME: asrc needs to use asrc_serial(spdif1) clock to do sample
 	 * rate convertion and this clock frequency can not be too high, set
-- 
1.7.9.5

